includes/Modules/AdSense.php

```diff
@@ -522,102 +525,43 @@ function( \Google_Service_AdSense_Alert $alert ) {
-					return $this->create_adsense_earning_data_request(
-						array(
-							'start_date' => date( 'Y-m-d', strtotime( 'yesterday' ) ),
-							'end_date'   => date( 'Y-m-d', strtotime( 'yesterday' ) ),
-						)
-					);
-				case 'earning-samedaylastweek':
-					return $this->create_adsense_earning_data_request(
-						array(
-							'start_date' => date( 'Y-m-d', strtotime( '7daysAgo' ) ),
-							'end_date'   => date( 'Y-m-d', strtotime( '7daysAgo' ) ),
-						)
-					);
-				case 'earning-7days':
-					return $this->create_adsense_earning_data_request(
-						array(
-							'start_date' => date( 'Y-m-d', strtotime( '7daysAgo' ) ),
-							'end_date'   => date( 'Y-m-d', strtotime( 'yesterday' ) ),
-						)
-					);
-				case 'earning-prev7days':
-					return $this->create_adsense_earning_data_request(
-						array(
-							'start_date' => date( 'Y-m-d', strtotime( '14daysAgo' ) ),
-							'end_date'   => date( 'Y-m-d', strtotime( '8daysAgo' ) ),
-						)
-					);
-				case 'earning-this-month':
-					return $this->create_adsense_earning_data_request(
-						array(
-							'start_date' => date( 'Y-m-01' ),
-							'end_date'   => date( 'Y-m-d', strtotime( 'today' ) ),
-						)
-					);
-				case 'earning-this-month-last-year':
-					$last_year          = intval( date( 'Y' ) ) - 1;
-					$last_date_of_month = date( 't', strtotime( $last_year . '-' . date( 'm' ) . '-01' ) );
-					return $this->create_adsense_earning_data_request(
-						array(
-							'start_date' => date( $last_year . '-m-01' ),
-							'end_date'   => date( $last_year . '-m-' . $last_date_of_month ),
-						)
-					);
-				case 'earning-28days':
-					return $this->create_adsense_earning_data_request(
-						array(
-							'start_date' => date( 'Y-m-d', strtotime( '28daysAgo' ) ),
-							'end_date'   => date( 'Y-m-d', strtotime( 'yesterday' ) ),
-						)
-					);
-				case 'earning-prev28days':
-					return $this->create_adsense_earning_data_request(
-						array(
-							'start_date' => date( 'Y-m-d', strtotime( '56daysAgo' ) ),
-							'end_date'   => date( 'Y-m-d', strtotime( '29daysAgo' ) ),
-						)
-					);
-				case 'earning-daily-this-month':
-					return $this->create_adsense_earning_data_request(
+				case 'earnings':
+					$data = array_merge(
 						array(
-							'dimensions' => array( 'DATE' ),
-							'start_date' => date( 'Y-m-01' ),
-							'end_date'   => date( 'Y-m-d', strtotime( 'today' ) ),
+							'dateRange'  => '',
+							'dimensions' => '',
 						)
 					);
-				case 'earnings-this-period':
-					$date_range = ! empty( $data['dateRange'] ) ? $data['dateRange'] : 'last-28-days';
-					switch ( $date_range ) {
-						case 'last-7-days':
-							$daysago = 7;
-							break;
-						case 'last-14-days':
-							$daysago = 14;
-							break;
-						case 'last-90-days':
-							$daysago = 90;
-							break;
-						case 'last-28-days':
-						default:
-							$daysago = 28;
-							break;
+
+					if ( empty( $data['dateRange'] ) ) {
```

Comment:
Can we instead have a default of `last-28-days` here, like most (all?) of the other datapoints do?

Reply:
Yes, I forgot we decided that this was not a required parameter 👍